### PR TITLE
Tracksight refactor 3

### PR DESCRIPTION
**Pull request by myung03**
### Changelist 
* Added path aliases for lib and ui (i.e to import particular ui component, import { Sonner } from @ts/ui)
* Restructured to move small ui components into a global package to be accessed from anywhere
* Added sonner, clsx, cva, tailwindmerge, tailwindanimate, shadcn, etc.
* refactor components to use sonner/toast instead of messageAPI
* Use Next.js built in navigation page structure
* Folders now contain page and associated component (visualize, dashboard, etc.), as well as any larger, non reusable components
* use ThemeProvider to handle light/dark mode
* update README
* small code style things to make code more readable
### Testing Done
* UI Testing to ensure that functionality remains the same. Only frontend changes were made; frontend logic/backend is unchaged. 
* Only Navbar has changed visually as navigation is now handled properly by Next.js.

### CODE REVIEW

File: software/tracksight/frontend/src/app/globals.css [hunk 1]
```diff
@@ -1,6 +1,75 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
```
Comment (Lucien950) [inline software/tracksight/frontend/src/app/globals.css:6]: duplicate?

File: software/tracksight/frontend/src/app/visualize/influxcomponents/query_data.tsx [hunk 2]
```diff
@@ -71,14 +65,13 @@ const QueryData = (props: QueryDataProps) => {
...
-        props.messageApi.open({ type: "error", content: error.toString() })
-      );
+      .then((data) => props.setData(data));
+    // TDOO replace with shadcn sonner
+    // .catch((error) => props.messageApi.open({ type: "error", content: error.toString() }));
   };
 
   return (
-    <div className="flex flex-col">
+    <div style={{ display: "flex", flexDirection: "column" }}>
```
Comment (Lucien950) [inline software/tracksight/frontend/src/app/visualize/influxcomponents/query_data.tsx:74]: tailwind?

**Review by Lucien950 — CHANGES_REQUESTED** [review]
love the new organization and the prettier formatting. Few more things before we merge; after this PR I think we should start looking towards next year and longer term to see what is in store there.

File: software/tracksight/frontend/src/app/navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+export default function NavBar() {
+  const { theme, setTheme } = useContext(ThemeContext);
+  const router = useRouter();
+  const path = usePathname();
+
+  return (
+    <>
+      <nav className="p-0 fixed top-0 left-0 right-0">
+        <Menu
```
[the remark this answers is not in the document]
Comment (myung03) [reply software/tracksight/frontend/src/app/navbar.tsx:16]: agree, I think that'll be one of the first things I work on while we start adding more features!

File: software/tracksight/frontend/tsconfig.json [hunk 4]
```diff
@@ -18,8 +18,9 @@
         "name": "next"
       }
     ],
+    "baseUrl": ".",
     "paths": {
```
[the remark this answers is not in the document]
Comment (myung03) [reply software/tracksight/frontend/tsconfig.json:22]: added, shortcut is "~"

File: software/tracksight/frontend/src/app/globals.css (hunk 1 above)
Comment (Lucien950) [inline software/tracksight/frontend/src/app/globals.css:6]:
```suggestion
```

**Review by Lucien950 — APPROVED** [review]

**Pull request merged**